### DISCUSSION TITLE
Why the gui sometimes doesn't allow creating only 1 recovery file ?

### DISCUSSION THREAD

**Discussion by uxmy** [Yutaka-Sawada/MultiPar#64] [Q&A] [answered]
Hi,
When i add multiple files as input, Multipar's gui usually forces me to create at least 2 , 3, sometimes 6 or even more recovery files (changing sizing scheme doesn't help). However, i can force the creation of only 1 recovery file using par2j from command line. Is there an explanation for this ?

**Comment by JohnLGalt** [comment]
Does the single recovery file have enough blocks to recover the missing / damaged file?

**Reply by uxmy** [reply]
> Does the single recovery file have enough blocks to recover the missing / damaged file?

Yes, the repair works perfectly. Here is the method to reproduce: 
1- Add 10 small files (like jpg or png images) to Multipar Gui
2- Set block size to: 32768
3- Set redundancy to: 100%
4- Try to reduce the number of recovery files to: 1 (you can't)
5- Create the recovery data anyway.
6- Move the recovery files to another partition
7- Open any recovery file from those
8- Click repair.

Now we try to make recovery data from the same input files, but with the command line:
1- Use a command like this:
par2j create -ss32768 -rr100 -rf1 randomename.par2 inputfile1 inputfile2 ....... inputfile10
2- Move the resulting recovery file to another partition, then open it.
3- Click repair.

All the files will be restored in both cases. But with the 2nd method, i think something different happens to the critical packets inside the recovery file. When i copy a part of the middle of the index file data (using a hex editor) , and i try to search for the same data inside the recovery file, i can't find repetitions. Maybe they are stored in a different way,  or there is only 1 copy of those critical packets.  If there is no repetitions, it's a problem.

**Comment by Yutaka-Sawada** [comment, marked as answer]
> Is there an explanation for this ?

Yes, the recovery file size is limited by default on MultiPar GUI. The default limit size is size of the maximum source file. It's same as "/lr" option at command-line.

For example, if you put 10 jpeg files of each 100 KB size, recovery files will be smaller than around 100 KB. To limit size of recovery file size, number of recovery files become multiple. This isn't a problem, because most users set small redundancy. You saw this strange as you set 100% redundancy.

You may set the limit size manually by specifying split size. When you check "Split Files" on MultiPar GUI, you can edit value of "Limit Size to". While the limit size is enabled, the size becomes the limit size of recovery files, too. Even after you uncheck "Split Files" and limit size is disabled, it won't re-calculate recovery file size again. So, you can set fewer recovery files by checking "Split Files" and uncheck later.


> Maybe they are stored in a different way, or there is only 1 copy of those critical packets. If there is no repetitions, it's a problem.

You can see how many packets in each recovery file. For example, there are below lines on par2j's output;

```
Loading PAR File       :
 Packet Slice Status   :  Filename
     26     0 Good     : "test1.par2"
    190    39 Good     : "test1.vol0+39.par2"
```

The first Index File contains 26 packets (25 critical packets and 1 creator packet). The second Recovery File contains 190 packets (150 critical packets and 1 creator packet and 39 recovery slice packets). So, I see 6 times of critical packets. Because 2 power 6 = 64, 39 blocks give 6 times. By reading output, you see how many repetition. If there is a failure, please report the creating time options and resulting output.

By the way, you may change "Packet repetition" on MultiPar Option -> "Client behavior" tab -> "Par2 options" section. The default setting is "No restriction" (power of 2). If you happens to set "Not used", MultiPar doesn't repeat packets. Normally the default setting is good. This is about "/lp" option at command-line.

**Reply by uxmy** [reply]
Thank you, Mr Yutaka for the detailed answer.
